bin/app.ts

```diff
@@ -4,87 +4,64 @@
  * Copyright 2023-2024 Amazon.com, Inc. or its affiliates.
  */
 
-import { App, Aspects, Environment } from "aws-cdk-lib";
+import { Aspects } from "aws-cdk-lib";
 import { AwsSolutionsChecks, NIST80053R5Checks } from "cdk-nag";
 
-import targetAccount from "../lib/accounts/target_account.json";
-import { deployDataCatalog } from "./deploy-data-catalog";
-import { deployDataIntake } from "./deploy-data-intake";
-import { deployModelRunner } from "./deploy-model-runner";
-import { deployRoles } from "./deploy-roles";
-import { deployTileServer } from "./deploy-tile-server";
-import { deployVpc } from "./deploy-vpc";
+import { deployDataCatalog } from "../lib/osml-stacks/data_catalog";
+import { deployDataIntake } from "../lib/osml-stacks/data_intake";
+import { deployModelRunner } from "../lib/osml-stacks/model_runner";
+import { deployRoles } from "../lib/osml-stacks/roles";
+import { deployTestImagery } from "../lib/osml-stacks/test_imagery";
+import { deployTestModelEndpoints } from "../lib/osml-stacks/test_model_endpoints";
+import { deployTileServerDataplane } from "../lib/osml-stacks/tile_server/ts_dataplane";
+import { deployVpc } from "../lib/osml-stacks/vpc";
+import { appConfig } from "./app_config";
 
-// Determine if the ENV instructs to globally build from source.
-const buildFromSource = process.env.BUILD_FROM_SOURCE?.toLowerCase() === "true";
-
-// Determine if we want to run CDK-Nag at Application Level
-const runCDKNagOnAppLevel = process.env.RUN_CDK_NAG?.toLowerCase() === "true";
-
-// Initialize the default CDK application.
-const app = new App();
-
-// Define the target AWS environment using account details from the configuration.
-const targetEnv: Environment = {
-  account: targetAccount.id,
-  region: targetAccount.region
-};
-
-// Deploy an optional role stack to build if we are deploying model runner.
+// Deploy an optional role stack
 let osmlRolesStack = undefined;
-if (targetAccount.deployModelRunner) {
-  osmlRolesStack = deployRoles(app, targetEnv, targetAccount);
+if (appConfig.modelRunner?.deploy) {
+  osmlRolesStack = deployRoles();
 }
 
 // Deploy required OSML networking infrastructure.
-const vpcStack = deployVpc(app, targetEnv, targetAccount, osmlRolesStack);
+const vpcStack = deployVpc(osmlRolesStack);
```

Comment:
I see a deploy boolean config in the cdk.context.json file for the VPC stack? Shouldn't we be conditionally deploying this?

Reply:
The VPC stack is not actually optional - it is required for any other resources to be deployed. But we do allow you to set a vpc_id so that it is imported into this construct rather than created from scratch.